e2etest/bgptests/bgp.go

```diff
@@ -879,145 +875,122 @@ var _ = ginkgo.Describe("BGP", func() {
-
-		table.DescribeTable("configure bgp local-preference and verify it gets propagated",
-			func(rangeWithPref, rangeNoPref string, family ipfamily.Family, localPref uint32) {
-				configData := config.File{
-					Pools: []config.AddressPool{
-						{
-							Name:      "pool-localpref",
-							Protocol:  config.BGP,
-							Addresses: []string{rangeWithPref},
-							BGPAdvertisements: []config.BgpAdvertisement{
-								{
-									LocalPref: uint32Ptr(localPref),
-								},
-							},
-						}, {
-							Name:      "pool-nolocalpref",
-							Protocol:  config.BGP,
-							Addresses: []string{rangeNoPref},
-						},
-					},
-					Peers: metallb.PeersForContainers(FRRContainers, family),
-				}
-				for _, c := range FRRContainers {
-					err := frrcontainer.PairWithNodes(cs, c, family)
-					framework.ExpectNoError(err)
-				}
-
-				err := ConfigUpdater.Update(configData)
-				framework.ExpectNoError(err)
+					validateService(cs, svcAdvertisement, allNodes.Items, c)
+					validateService(cs, svcAdvertisement1, allNodes.Items, c)
+					validateService(cs, svcNoAdvertisement, allNodes.Items, c)
+					Eventually(func() error {
+						for _, community := range advertisement.Communities {
+							routes, err := frr.RoutesForCommunity(c, community, ipFamily)
+							if err != nil {
+								return err
+							}
+							if _, ok := routes[ipNoAdvertisement]; ok {
+								return fmt.Errorf("found %s route for community %s", community, ipNoAdvertisement)
+							}
+							if _, ok := routes[ipWithAdvertisement1]; !ok {
+								return fmt.Errorf("%s route not found for community %s", community, ipWithAdvertisement1)
+							}
+							if _, ok := routes[ipWithAdvertisement]; !ok {
+								return fmt.Errorf("%s route not found for community %s", community, ipWithAdvertisement)
+							}
+						}
+						// LocalPref check is only valid for iBGP sessions
+						if advertisement.LocalPref != nil && strings.Contains(c.Name, "ibgp") {
+							localPrefix, err := frr.LocalPrefForPrefix(c, ipWithAdvertisement, ipFamily)
+							if err != nil {
+								return err
+							}
+							if localPrefix != *advertisement.LocalPref {
+								return fmt.Errorf("%s %s not matching local pref", c.Name, ipWithAdvertisement)
+							}
+							localPrefix, err = frr.LocalPrefForPrefix(c, ipWithAdvertisement1, ipFamily)
+							if err != nil {
+								return err
+							}
+							if localPrefix != *advertisement.LocalPref {
+								return fmt.Errorf("%s %s not matching local pref", c.Name, ipWithAdvertisement1)
+							}
+							localPrefix, err = frr.LocalPrefForPrefix(c, ipNoAdvertisement, ipFamily)
+							if err != nil {
+								return err
+							}
+							if localPrefix == *advertisement.LocalPref {
+								return fmt.Errorf("%s %s matching local pref", c.Name, ipNoAdvertisement)
+							}
 
-				for _, c := range FRRContainers {
-					validateFRRPeeredWithNodes(cs, c, family)
+						}
+						return nil
+					}, 1*time.Minute, 1*time.Second).Should(BeNil())
```

Comment:
1 min for this validation is enough?

Reply:
Looks like. Routes are already set and we run validateService before